packages/next/src/server/lib/incremental-cache/index.ts

```diff
@@ -556,7 +559,11 @@ export class IncrementalCache implements IncrementalCacheType {
 
     if (this.dev && !ctx.fetchCache) return
     // fetchCache has upper limit of 2MB per-entry currently
-    if (ctx.fetchCache && JSON.stringify(data).length > 2 * 1024 * 1024) {
+    if (
+      ctx.fetchCache &&
+      this.isFetchCacherPossible &&
+      JSON.stringify(data).length > 2 * 1024 * 1024
+    ) {
       if (this.dev) {
         throw new Error(`fetch for over 2MB of data can not be cached`)
       }
```

Comment:
How about we move this check into the `fetch-cache` interface itself?

Reply:
Good idea, but there is one thing I don't like about it:

Let's assume that we are in development mode and looking at the same services as in production. Everything works fine, we build and test locally - it works well.

Then we deploy to Vercel and strange behavior starts - I described it in the discussion here #60129. We can't replicate it locally, there are no logs (at the very least, we need to add these logs, but the question is, is it normal to have different behavior in the built application? And if it goes unnoticed, how harmful is it?).

It may be worth moving the error inside, but leaving an informer here that if you use Vercel, this request cannot be cached because it is larger than 2MB and, perhaps, write about the error somewhere in the doc.

Reply:
That's a good point, it is valuable having this warning in dev so it's not only a production error. Sounds good to leave here for now